### ISSUE TITLE
"dotnet add package" should commit the restore it performs to the assets file

### ISSUE THREAD

**Issue by anangaur** [NuGet/Home#6928]
When I run `dotnet add package`, the package is not restore although it mentions so as a log info:
```
C:\Play\Nuget\dotnet\sample\projectB>dotnet add package newtonsoft.json
  Writing C:\Users\anangaur\AppData\Local\Temp\tmpFFE3.tmp
info : Adding PackageReference for package 'newtonsoft.json' into project 'C:\Play\Nuget\dotnet\sample\projectB\projectB.csproj'.
log  : Restoring packages for C:\Play\Nuget\dotnet\sample\projectB\projectB.csproj...
info :   CACHE https://api.nuget.org/v3-flatcontainer/newtonsoft.json/index.json
info : Package 'newtonsoft.json' is compatible with all the specified frameworks in project 'C:\Play\Nuget\dotnet\sample\projectB\projectB.csproj'.
info : PackageReference for package 'newtonsoft.json' version '11.0.2' added to file 'C:\Play\Nuget\dotnet\sample\projectB\projectB.csproj'.
```

We should replace the following
`log  : Restoring packages for C:\Play\Nuget\dotnet\sample\projectB\projectB.csproj...`
with
`log  : Computing packages' graph for C:\Play\Nuget\dotnet\sample\projectB\projectB.csproj...` or equivalent that does not tell users that restore has taken place.

**Comment by jainaashish** [comment]
dotnet add package actually does a restore but doesn't commit the restore artifacts so the right fix is to commit those artifacts.

**Comment by anangaur** [comment]
👍🏽

**Comment by jainaashish** [comment]
Merged for 15.9 into release-4.9.0-preview2 branch with commit# cbdd386e3b9c3053e57385efc73d0facd9df4600
